styles/learnxinyminutes/catppuccin.user.css

```diff
@@ -0,0 +1,250 @@
+            letter-spacing: 0.1em;
+            margin-left: 0.05em;
+            margin-right: 0.05em;
+        } // Code blocks
+        .highlight {
+            background-color: @mantle;
+
+            .c1, // inline comments
+            .cm, // block comments
+            .c, // css comments
+            .p, // braces / delimiters
+            {
+                color: @overlay2 !important;
+            }
+
+            .se, // escape sequences
+            .sc {
+                color: @pink !important;
+            }
+
+            .kt, // types
+            .ss,
+            .mo,
+            .vg,
+            .ne, // errors
+            .mh,
+            .nt, // element selector (css)
+            .mb,
+            .kp, // & in rust
+            {
+                color: @yellow !important;
+            } // Variables
+            .n,
+            .nc,
+            .nx,
+            .gh, // markdown h1
+            .gu, // markdown h2-h6
+            .gs, // markdown italics
+            .nv,
+            .nn {
+                color: @text !important;
+            }
+
+            .ge // markdown bold
+            {
+                color: @text !important;
+            }
+
+            .gd // markdown strikethrough
+            {
+                color: @text !important;
+                background-color: @mantle !important;
+                text-decoration: line-through;
+            }
+
+            .sd, // Documentation comments
+            .cp, // Directives
+            {
+                color: @pink !important;
+            } // Operators
+            .o, // operators
+            .na, // html attribute names
+            .nl, // markdown reference links
+            .ow {
+                color: @sky;
+            } // Constants, numbers and e.g. print()
+            .mi,
+            .m,
+            .nd, // decorators
+            .kc,
+            .mf {
+                color: @peach !important;
+            }
+
+            // This one will appear when their highlighter library can't parse certain text. This only happens in .scss files where $ will have this class
+            .err {
+                color: @sky;
+                background-color: @mantle;
+            }
+
+            .s1, // inline strings
+            .s2, // block strings
+            .s,
+            .sd, // multiline strings
+            .sh, // multiline strings
+            .sx, // url strings in css
+            .sb {
+                color: @green !important;
+            } // Control flow and variables within template literals
+            .k,
+            .si {
+                color: @red !important;
+            } // Keywords
+            .kd,
+            .k,
+            .kr,
+            .kn, // module syntax
+            .no, // atom in lisp
+            .nb,
+            .fm {
+                color: @mauve !important;
+            } // Functions
+            .nf {
+                color: @blue !important;
+            }
+        }
```

Comment:
Please see https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md#pygments. You can import our existing port for the syntax highlighter used (Pygments) instead of theming it here.

Reply:
> Please see https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md#pygments. You can import our existing port for the syntax highlighter used (Pygments) instead of theming it here.

I tried it but it doesn't look right

Before:

![image](https://github.com/user-attachments/assets/fe5bed10-2de0-48c7-8102-facf5a7f6201)


After:

![image](https://github.com/user-attachments/assets/79a10478-6c70-4cf8-bf58-3e292a147b43)

My code editor with the same code: (i use catppuccin theme, with zero modifications)
![image](https://github.com/user-attachments/assets/4edfea2d-2825-4e86-bdc3-61cb10f91d81)

Reply:
That would be issues with the upstream port. Let's still go with it, and we can open an issue there. I'll review it later this week along with another port I need to review token highlighting for.

Reply:
> That would be issues with the upstream port. Let's still go with it, and we can open an issue there. I'll review it later this week along with another port I need to review token highlighting for.

Where is the origin of that file on github so I can send a PR to fix? I have effectively created a version for pygments which aligns closer to what catppuccin really should look like, so I think a good starting point would be to merge my addition with what is already there

also - some considerations. Using `import` and sending an extra network request, is it really worth it? Why not just have people copy-paste everything into the userstyle. (since they'd be copy-pasting the css variables either way)

Reply:
> Where is the origin of that file on github so I can send a PR to fix? I have effectively created a version for pygments which aligns closer to what catppuccin really should look like, so I think a good starting point would be to merge my addition with what is already there

I've raised a tracking issue: https://github.com/catppuccin/python/issues/72. It's not as simple as copy pasting CSS class names, as it is generated differently.

> also - some considerations. Using `import` and sending an extra network request, is it really worth it? Why not just have people copy-paste everything into the userstyle. (since they'd be copy-pasting the css variables either way) otherwise each site which imports that file will load slightly slower

It is worth it because having so many different syntax highlighting implementations across all our (100) userstyles is infeasible. Nobody wants to update that. The network request is negligible on modern hardware, and we already require the latest versions of web browsers so it is a reasonable expectation.